pkg/collect/copy_from_host.go

```diff
@@ -424,3 +430,34 @@ func deleteDaemonSet(client kubernetes.Interface, ctx context.Context, createdDS
 		}
 	}
 }
+
+func checkDaemonsePodStatus(client kubernetes.Interface, ctx context.Context, labels map[string]string, namespace string) error {
+	var labelSelector []string
+	for k, v := range labels {
+		labelSelector = append(labelSelector, fmt.Sprintf("%s=%s", k, v))
+	}
+	pods, err := client.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: strings.Join(labelSelector, ","),
+	})
+	if err != nil {
+		return errors.Wrap(err, "get daemonset pods")
+	}
+
+	for _, pod := range pods.Items {
+		if pod.Status.Phase != corev1.PodRunning {
+			events, _ := client.CoreV1().Events(namespace).List(ctx, metav1.ListOptions{
+				FieldSelector: fmt.Sprintf("involvedObject.uid=%s", pod.UID),
+			})
+
+			for _, event := range events.Items {
+				// If the pod has a FailedMount event, it means that the pod failed to mount the volume and the pod will be stuck in the Pending state.
+				// In this case, we return an error to the caller to indicate that path does not exist.
+				if event.Reason == "FailedMount" {
```

Comment:
I would prefer if this decision was driven by an argument passed by the caller (not spec). We can add a `RetryFailedMount` to `CollectCopyFromHost` which once set to `true` will have the collector behave the same way as it is today. I say this cause the collector can be used other scenarios in addition to `collectd`. 

`collectd`: `RetryFailedMount=false`
Default: `RetryFailedMount=true`

Reply:
Good idea, there is a issue which I need some help. One thing is that how we can set RetryFailedMount as defaul true if the user not defined it in yaml
we are using 
```
type Collectd struct {
        .....
	RetryFailedMount      bool             `json:"retryFailedMount,omitempty" yaml:"retryFailedMount,omitempty"`
}
```
since it is bool, as default, it will be false. We don't have a clear way to identify whether the user has set it to be false, or just didn't set it. 

I can switch it to string, however, that will bring another issue that we need to use "true" instead of true.

I updated the pr with introducing a argument called retryLimit. If the event count has reached retry limit, which means that k8s retried few times but it didn't work. I think that is a good signal that we should terminate the pod instead of timeout.

In default, without defined in yaml, retryLimit will be 5.

Reply:
My suggestion was to add it in the `CollectCopyFromHost` struct, not the spec like so

https://github.com/banjoh/troubleshoot/commit/4e0b359525ec37bdd8c9d7392c81f7fd4ba486d5

Reply:
Sorry, I missed the (not spec) part. Now the PR has been updated to add it in the CollectCopyFromHost